addons/mail/models/mail_mail.py

```diff
@@ -395,21 +399,39 @@ def _prepare_outgoing_list(self):
                 link_ids = {int(link) for link in re.findall(r'/web/(?:content|image)/([0-9]+)', body)}
                 if link_ids:
                     attachments = attachments - self.env['ir.attachment'].browse(list(link_ids))
+
+            # Compute email size and transform attachment in downloadable link in the body if it exceeds the limit
+            estimated_email_size = (
+                    (len(json.dumps(headers, ensure_ascii=False).encode(errors='ignore')) +
+                     (len(body.encode(errors='ignore')) if body else 0) +
+                     sum(attachment['file_size']
+                         for attachment in attachments.sudo().read(['file_size']))) * BASE_64_SIZE_OVERHEAD_RATIO +
```

Comment:
Can't we read attachments before, and then compute email size and email_attachments based on it ? That way we don't read it twice (even if I suppose the cache is working ?)